PWGHF/TableProducer/candidateSelectorXicToXiPiPi.cxx

```diff
@@ -0,0 +1,247 @@
+        registry.get<TH2>(HIST("hSelections"))->GetXaxis()->SetBinLabel(iBin + 1, labels[iBin].data());
+      }
+    }
+  }
+
+  /// Conjugate-independent topological cuts
+  /// \param candidate is candidate
+  /// \return true if candidate passes all cuts
+  template <typename T1>
+  bool selectionTopol(const T1& hfCandXic)
+  {
+    auto candpT = hfCandXic.pt();
+    int pTBin = findBin(binsPt, candpT);
+    if (pTBin == -1) {
+      return false;
+    }
+
+    // check that the candidate pT is within the analysis range
+    if (candpT < ptCandMin || candpT >= ptCandMax) {
+      return false;
+    }
+
+    // check candidate mass is within a defined mass window
+    if (std::abs(hfCandXic.invMassXic() - o2::constants::physics::MassXiCPlus) > cuts->get(pTBin, "m")) {
+      return false;
+    }
+
+    // cosine of pointing angle
+    if (hfCandXic.cpa() <= cuts->get(pTBin, "cos pointing angle")) {
+      return false;
+    }
+
+    // cosine of pointing angle XY
+    if (hfCandXic.cpaXY() <= cuts->get(pTBin, "cos pointing angle XY")) {
+      return false;
+    }
+
+    // candidate maximum decay length
+    if (hfCandXic.decayLength() > cuts->get(pTBin, "max decay length")) {
+      return false;
+    }
+
+    // candidate maximum decay length XY
+    if (hfCandXic.decayLengthXY() > cuts->get(pTBin, "max decay length XY")) {
+      return false;
+    }
+
+    // candidate chi2PC
+    if (hfCandXic.chi2PCA() > cuts->get(pTBin, "chi2PCA")) {
+      return false;
+    }
+
+    // maximum DCA of daughters
+    if ((std::abs(hfCandXic.impactParameter0()) > cuts->get(pTBin, "max impParXY Xi")) ||
+        (std::abs(hfCandXic.impactParameter1()) > cuts->get(pTBin, "max impParXY Pi0")) ||
+        (std::abs(hfCandXic.impactParameter2()) > cuts->get(pTBin, "max impParXY Pi1"))) {
+      return false;
+    }
+
+    // cut on daughter pT
+    if (hfCandXic.ptProng0() < cuts->get(pTBin, "pT Xi") ||
+        hfCandXic.ptProng1() < cuts->get(pTBin, "pT Pi0") ||
+        hfCandXic.ptProng2() < cuts->get(pTBin, "pT Pi1")) {
+      return false;
+    }
+
+    return true;
+  }
+
+  /// Apply PID selection
+  /// \param pidTrackPi0   PID status of trackPi0 (prong1 of Xic candidate)
+  /// \param pidTrackPi1   PID status of trackPi1 (prong2 of Xic candidate)
+  /// \param pidTrackPr    PID status of trackPr (positive daughter of V0 candidate)
+  /// \param pidTrackPiLam PID status of trackPiLam (negative daughter of V0 candidate)
+  /// \param pidTrackPiXi  PID status of trackPiXi (Bachelor of cascade candidate)
+  /// \param acceptPIDNotApplicable switch to accept Status::NotApplicable
+  /// \return true if prong1 and prong 2 of Xic candidate pass all selections
+  template <typename T1, typename T2, typename T3, typename T4, typename T5, typename T6 = bool>
+  bool selectionPid(const T1& pidTrackPi0, const T2& pidTrackPi1, const T3& pidTrackPr, const T4& pidTrackPiLam, const T5& pidTrackPiXi, const T6& acceptPIDNotApplicable)
+  {
```

Comment:
Why is this templated? You always call the method with the same types. Also there is no benefit passing by reference.